pkg/asset/agent/joiner/clusterinfo.go

```diff
@@ -254,6 +254,16 @@ func (ci *ClusterInfo) retrieveArchitecture() error {
 func (ci *ClusterInfo) retrieveFIPS() error {
 	workerMachineConfig, err := ci.OpenshiftMachineConfigClient.MachineconfigurationV1().MachineConfigs().Get(context.Background(), "99-worker-fips", metav1.GetOptions{})
 	if err != nil {
+		// Older oc client may not have sufficient permissions,
+		// falling back to previous implementation.
+		if errors.IsForbidden(err) {
+			installConfig, err := ci.retrieveInstallConfig()
+			if err != nil {
+				return err
+			}
+			ci.FIPS = installConfig.FIPS
```

Comment:
retrieveInstallConfig can return `nil, nil` at line 351.
Seems that you need to check installConfig != nil before using it

Reply:
Nice catch, thanks @bfournie !
Fixed as per previous implementation